test/geometry/epipolar/test_essential.py

```diff
@@ -8,6 +10,103 @@
 from kornia.testing import assert_close
 
 
+class TestFindEssential:
+    def test_smoke(self, device, dtype):
+        points1 = torch.rand(1, 5, 2, device=device, dtype=dtype)
+        points2 = torch.rand(1, 5, 2, device=device, dtype=dtype)
+        weights = torch.ones(1, 5, device=device, dtype=dtype)
+        E_mat = epi.essential.find_essential(points1, points2, weights)
+        assert E_mat.shape == (1, 3, 3)
+
+    @pytest.mark.parametrize("batch_size, num_points", [(1, 5), (2, 6), (3, 7)])
+    def test_shape(self, batch_size, num_points, device, dtype):
+        B, N = batch_size, num_points
+        points1 = torch.rand(B, N, 2, device=device, dtype=dtype)
+        points2 = torch.rand(B, N, 2, device=device, dtype=dtype)
+        weights = torch.ones(B, N, device=device, dtype=dtype)
+        E_mat = epi.essential.find_essential(points1, points2, weights)
+        assert E_mat.shape == (B, 3, 3)
+
+    @pytest.mark.parametrize("batch_size, num_points", [(1, 5), (2, 6), (3, 7)])
+    def test_shape_noweights(self, batch_size, num_points, device, dtype):
+        B, N = batch_size, num_points
+        points1 = torch.rand(B, N, 2, device=device, dtype=dtype)
+        points2 = torch.rand(B, N, 2, device=device, dtype=dtype)
+        weights = None
+        E_mat = epi.essential.find_essential(points1, points2, weights)
+        assert E_mat.shape == (B, 3, 3)
+
+    def test_epipolar_constraint(self, device, dtype):
+        scene: Dict[str, torch.Tensor] = utils.generate_two_view_random_scene(device, dtype)
+        x1 = scene['x1'][:, :5, :]
+        x2 = scene['x2'][:, :5, :]
+        K1 = scene['K1']
+        K2 = scene['K2']
+
+        try:
+            import cv2
```

Comment:
i don't think it's a good idea. We prefer to adopt hardcoded cases for cases which should match with cv2, numpy, etc
cc @edgarriba

Reply:
Yeah, generate your test samples and hardcode

Reply:
I see, it's not a good test. I was supposed to use the calibration method from Kornia to normalized the points for E estimation, but failed because there is no distortion coefficient from the [generated two-view random scene](https://github.com/weitong8591/kornia/blob/solvers/essential/test/geometry/epipolar/test_common.py), so I change the calibration to OpenCV version. How should I do it in a correct way?

Reply:
> Yeah, generate your test samples and hardcode

I see! I just saw this reply after I commented. I will fix it. Thanks!